uaclient/cli.py

```diff
@@ -1073,22 +1083,72 @@ def action_auto_attach(args, *, cfg):
 @assert_root
 @assert_lock_file("ua attach")
 def action_attach(args, *, cfg):
-    if not args.token:
+    if not args.token and not args.attach_config:
         raise exceptions.UserFacingError(
             ua_status.MESSAGE_ATTACH_REQUIRES_TOKEN
         )
-    try:
-        actions.attach_with_token(
-            cfg, token=args.token, allow_enable=args.auto_enable
+    if args.token and args.attach_config:
+        raise exceptions.UserFacingError(
+            ua_status.MESSAGE_ATTACH_TOKEN_ARG_XOR_CONFIG
         )
+
+    enable_services_override = None
+    if args.token:
+        token = args.token
+    else:
+        try:
+            attach_config = AttachActionsConfigFile.from_dict(
+                yaml.safe_load(args.attach_config)
```

Comment:
Should we have an exception in case the file doesn't exist ?

Reply:
Oh probably, that might happen during the argparse-ing though. I'll play with this.

Reply:
@lucasmoura Right now the following happens
```
root@ua-xenial-e5fbbc69376e12d37d73cd72114de337:~# ua attach --attach-config=/tmp/nonexistent.yaml
usage: ua attach <token> [flags]
argument --attach-config: can't open '/tmp/nonexistent.yaml': [Errno 2] No such file or directory: '/tmp/nonexistent.yaml'
```
What do you think?

Reply:
Awesome, I think we can keep as is then